### ISSUE TITLE
Swallow(Task task) is not available in Xamarin.iOS and Xamarin.Droid. 

### ISSUE THREAD

**Issue by demogog** [NLog/NLog#3703]
Hi! Is there any reason why `Swallow(Task task)` method is absent from Xamarin.iOS and Xamarin.Droid?
To my best knowledge, Tasks are very well supported in both Xamarin platforms. 

I currently referenced a .NETStandard2.0 library in my Xamarin projects. But I get compile time error like this: 
`MTOUCH : error MT2101: Can't resolve the reference 'System.Void NLog.Logger::Swallow(System.Threading.Tasks.Task)', referenced from the method 'System.Void MobiCloudServices.Utils.Logging.Logger::Swallow(System.Threading.Tasks.Task)' in 'NLog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c'.`

From what I can see in NLog Logger source code [https://github.com/NLog/NLog/blob/6a00ed489a137d8f85b12620517ee585da36e429/src/NLog/Logger.cs#L502](url) the Swallow methods with Task parameter are stripped out by compiler directive. 

Is this a bug or intended behaviour? 

**NLog version**: 4.6.8

**Platform**: Xamarin.iOS / Xamarin.Droid / .NETStandard 2.0

**Comment by snakefoot** [comment]
NET4_5 includes all platforms that supports Net45-Framework-API. This includes NetStandard1 + NetStandard2.

But it doesn't include monoandroid44 + xamarinios10. Seems your project is not using the NetStandard-build of NLog.dll.

**Comment by demogog** [comment]
Hi Rolf. My project has to use Xamarinios10, since it is an iOS App - it can't be NETStandard2. 
However, the project can consume NETStandard2 libraries. 
The nuspec file of NLog states that iOS projects has to reference Xamarin dll and not a NETStandard dll. 
But that is not the point of this thread. 
My original question was - why swallow(task task) is not available on Xamarin, since Xamarin.iSO10 and Monodroid44 both support tasks. 
In other words, why are the `#if NET4_5` directive there? Maybe there should be corresponding directives for IOS and Droid, since they both support Tasks?

**Comment by snakefoot** [comment]
> In other words, why are the #if NET4_5 directive there? Maybe there should be corresponding directives for IOS and Droid, since they both support Tasks?

If MonoAndroid44 + XamarinIOs10 was removed as target platform for NLog 5.0 would it also work for you? (Leaving NetStandard1.3 + NetStandard1.5 + NetStandard2.0)

**Comment by demogog** [comment]
> Because Xamarin Mobile Platform and Silverlight Platform are treated as similar limited platforms. Silverlight only supports Net40 (without Tasks).

Okay, that explains it. I forgot about Silverlight, since it is deprecated almost 2 years ago. 

> If MonoAndroid44 + XamarinIOs10 was removed as target platform for NLog 5.0 would it also work for you? (Leaving NetStandard1.3 + NetStandard1.5 + NetStandard2.0)

For instance, Microsoft.indentitymodel.Tokens package has net45, net451, net461, netstandards1.4 and netstandard2.0 as supported target framework. So to answer your question, if monoandroid44 and xamarinios10 was removed, it would still work for Xamarin platform.

**Comment by snakefoot** [comment]
Closing this question as duplicate of #2871 (When there is only .NET Framework and .NetStandard then it will be resolved).
